json.go

```diff
@@ -137,3 +139,12 @@ func SyntaxErrorOffset(err error) (isSyntaxError bool, offset int64) {
 		return false, 0
 	}
 }
+
+// FieldError is an error that provides access to the path of the erroneous field
+type FieldError interface {
+	error
+	// FieldPath provides the full path of the erroneous field within the json object.
+	FieldPath() string
+	// SetFieldPath updates the path of the erroneous field to the one provided as `path`.
```

Comment:
tweak to this to make it clear this only impacts the error message:

```suggestion
	// SetFieldPath updates the path of the erroneous field output in the error message
```

Reply:
done